frontend/src/scenes/insights/insightDataLogic.ts

```diff
@@ -250,19 +256,20 @@ export const insightDataLogic = kea<insightDataLogicType>([
          * that haven't been refactored to use the data exploration yet.
          */
         insightData: (insightData: Record<string, any> | null) => {
-            if (!values.isUsingDataExploration) {
+            if (!values.isUsingDataExploration || insightData === null) {
                 return
             }
 
-            actions.setInsight(
-                {
-                    ...values.insight,
-                    result: insightData?.result,
-                    next: insightData?.next,
-                    filters: values.insight.query ? {} : queryNodeToFilter(values.querySource),
-                },
-                {}
-            )
+            // todo what about results that aren't in this shape?
```

Comment:
@thmsobrmlr to check I'm not missing things here ... 

spreading `values.insight` and calling `setInsight` here was collapsing the inflightFilters and things.

I think all we need to do is update the results we just received back but....